data_sources/create-views.sql

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2016 FCP.
+ *
+ * Requires a DB to be created.
+ *
+ * PostgreSQL cheat-sheet:
+ * - Launch console: psql
+ * - Execute script: \i <name of this file>.sql
+ * - Show all DB's: \l
+ * - Show tables: \dt
+ * - Describe table: \d FCP.<tablename>
+ * - Drop the whole schema (e.g., for tests):
+ *    drop schema FCP cascade;
+*/
+
+CREATE MATERIALIZED VIEW FCP.recipeingredientsproperties
+AS SELECT
+recipeingredients.recipe_id,
+ingredients.ingredient_id,
+recipeingredients.ingredient,
+recipeingredients.quantity,
+recipeingredients.unit_of_measure,
+recipeingredients.preparation_technique,
+recipeingredients.alternate_ingredient,
+ingredients.properties,
+ingredients.allergene
+FROM FCP.recipeingredients recipeingredients, FCP.ingredients ingredients
+WHERE recipeIngredients.ingredient = ingredients.name ;
+
+
+CREATE MATERIALIZED VIEW FCP.recipeeventsnames
+AS SELECT
+recipeevents.recipe_id,
+recipeevents.event_id,
+events.name,
+events.place,
+events.start_date,
+events.end_date,
+events.category
+
+FROM FCP.recipeevents recipeevents, FCP.events events
+WHERE recipeevents.event_id = events.event_id ;
```

Comment:
Al volo:

- al momento la creazione delle view e' statica; percio' la faccio dopo
aver riempito le tabelle (vedi il file insert_all.bat). Se la vuoi fare
prima devi assicurarti che la view si auto-updati quando carichi le tabelle
con qualche opzione di auto-refresh, verificando che funzioni (e qui mi
sovviene il proverbio texano: If it ain't broke... don't fix it ;-) )

- si', is_main e' un campo ridondante, infatti di certo non lo uso, pensavo
pure di averlo tolto...


2016-09-07 18:18 GMT+02:00 Claudio Guidi <notifications@github.com>:

> In data_sources/create-views.sql
> <https://github.com/FoodConnectsPeople/foodconnectspeople-tool/pull/10#discussion_r77855780>
> :
>
> > +FROM FCP.recipeingredients recipeingredients, FCP.ingredients ingredients
> > +WHERE recipeIngredients.ingredient = ingredients.name ;
> > +
> > +
> > +CREATE MATERIALIZED VIEW FCP.recipeeventsnames
> > +AS SELECT
> > +recipeevents.recipe_id,
> > +recipeevents.event_id,
> > +events.name,
> > +events.place,
> > +events.start_date,
> > +events.end_date,
> > +events.category
> > +
> > +FROM FCP.recipeevents recipeevents, FCP.events events
> > +WHERE recipeevents.event_id = events.event_id ;
>
> Io metterei questi script direttamente dentro il create-schema.sql così
> almeno quando creiamo il db creiamo anche le view.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FoodConnectsPeople/foodconnectspeople-tool/pull/10/files/4c89bf11de21d87125ebb2939eded8bd64b972a2#r77855780>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATvW3gNEnHGrLMAtfxSOcwr8g7YmQXcXks5qnuO-gaJpZM4J2MX6>
> .
>